pkg/controller/replication_controller.go

```diff
@@ -221,6 +221,13 @@ func (rm *ReplicationManager) syncReplicationController(controller api.Replicati
 		}
 		wait.Wait()
 	}
+	if controller.Status.Replicas != controller.Spec.Replicas {
```

Comment:
missing push? The if statement should match the line below, no?

Can you add/modify a test case so that this gets tested with 0 < len(filteredList) < spec.Replicas?